### PR TITLE
power-assert: update assert function

### PR DESCRIPTION
**Pull request by falsandtru**
Use http://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions

### CODE REVIEW

File: types/power-assert/ts3.7/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import * as module from "../index";
+
+export = assert;
+export as namespace assert;
+
+declare function assert(value: any, message?: string): asserts value;
+declare namespace assert {
+    export import AssertionError = module.AssertionError;
+
+    export import fail = module.fail;
```
Comment (rbuckton) [inline types/power-assert/ts3.7/index.d.ts]: Would it make more sense to have fail be the following?
```suggestion
    export function fail(actual?: any, expected?: any, message?: string, operator?: string): never;    
```

File: types/power-assert/ts3.7/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    export import ok = module.ok;
```
Comment (rbuckton) [inline types/power-assert/ts3.7/index.d.ts]: Would it make more sense to have `ok` be the following?
```suggestion
    export function ok(value: any, message?: string): asserts value;
```

File: types/power-assert/ts3.7/index.d.ts (hunk 1 above)
Comment (falsandtru) [reply types/power-assert/ts3.7/index.d.ts]: Why? All the others are the same?

Comment (falsandtru) [reply types/power-assert/ts3.7/index.d.ts]: I didn't see the return type. I'll fix the original definition.

Comment (falsandtru) [reply types/power-assert/ts3.7/index.d.ts]: done

File: types/power-assert/ts3.7/index.d.ts (hunk 2 above)
Comment (falsandtru) [reply types/power-assert/ts3.7/index.d.ts]: done.

**Review by rbuckton — APPROVED** [review]

**Pull request merged**